packages/scaffold/src/views/w3m-account-settings-view/index.ts

```diff
@@ -236,6 +243,13 @@ export class W3mAccountSettingsView extends LitElement {
 
     this.loading = true
     await emailConnector?.provider.setPreferredAccount(accountTypeTarget)
+    await ConnectionController.reconnectExternal(emailConnector)
+    this.switched = true
+    this.text =
+      this.text === 'Switch to your smart account'
```

Comment:
Why don't we use `this.preferredAccountType === W3mFrameRpcConstants.ACCOUNT_TYPES.SMART_ACCOUNT` as we did on top? Using hard coded string for condition doesn't seem reliable. We should be avoiding as much as possible.